action/protocol/staking/delegate.go

```diff
@@ -7,16 +7,28 @@
 package staking
 
 import (
-	"bytes"
 	"math/big"
 	"sort"
+	"strings"
 
 	"github.com/golang/protobuf/proto"
-	"github.com/iotexproject/go-pkgs/hash"
 	"github.com/iotexproject/iotex-address/address"
 	"github.com/pkg/errors"
 
+	"github.com/iotexproject/iotex-core/action/protocol"
 	"github.com/iotexproject/iotex-core/action/protocol/staking/stakingpb"
+	"github.com/iotexproject/iotex-core/state"
+	"github.com/iotexproject/iotex-core/state/factory"
+)
+
+/*
+ * keys in namespace factory.DelegateNameSpace are prefixed with 1-byte tag
+ * to indicate the type of stored object. This enables to quickly scan the
+ * entire bucket and filter out objects of a certain type
+ */
+const (
+	inActiveDel = byte(iota)
+	activeDel   = inActiveDel + 1
```

Comment:
could you elaborate or comment on the usage of these two parameters?

Reply:
prefix for storing inactive and active delegates, see stakingPutDelegate()

so on Start(), can quickly scan the entire bucket and filter out active delegates